src/plugins/cosmos/components/modals/Staking/views/Stake.tsx

```diff
@@ -96,9 +119,28 @@ export const Stake = ({ assetId, apr }: StakeProps) => {
       .times(_percent)
       .dp(asset.precision, BigNumber.ROUND_DOWN)
     const fiatAmount = bnOrZero(cryptoAmount).times(marketData.price)
+    const fiatAmountMinusTxFee = fiatAmount.minus(averageTxFee?.fiatFee ?? '0')
+    const cryptoAmountMinusTxFee = cryptoAmount.minus(averageTxFee?.txFee ?? '0')
+    const shouldSubtractFees = cryptoAmount
```

Comment:
Not following this one. Why would we not subtract fees?

Reply:
We don't always want to subtract the fees, only in the case where the range is between (max wallet balance - 0.0035) and max wallet balance. Other than that, we should still *check* for the fees to know if the Tx can be sent, but given there is a valid balance for that Tx to be broadcasted, there is no need to actually subtract them

Reply:
Nice terminology, renamed `fiatAmountMinusTxFee` and `cryptoAmountMinusTxFee` to `maxFiatStakeAmount` and `maxCryptoStakeAmount`.

Not sure about renaming `shouldSubtractFees`, while it indicates well what's being done, I agree it doesn't indicate well **why** it's doing it - is there any naming you have in mind for this one specifically, or is that rather its logic you'd see changed?

Reply:
I guess it works it just feels like a weird way of doing it.

Its a lot simpler for me to think about if we do something like:

if(currentBalance < expectedFees) insufficientFunds
if(stakeAmount > maxCryptoStakeAmount) stakeAmount = maxCryptoStakeAmount

Reply:
Nice, yeah I like this version, it's way more readable. Also allowed me to visually separate and comments the 3 steps of the process (get initial amount from input/ check for the max possible amount/ use user input or default to max)